### ISSUE TITLE
Filesystem "proc" must be mounted on ordinary directory

### ISSUE THREAD

**Issue by sneal** [vito/oci-build-task#35]
Concourse 6.40

I'm using the oci-image-task to attempt to build an image, however while executing a RUN command in my Dockerfile it gets this error:

```
container_linux.go:367: starting container process caused: process_linux.go:459: container init caused: rootfs_linux.go:58: mounting "proc" to rootfs at "/proc" caused: filesystem "proc" must be mounted on ordinary directory
```

The error seems to be coming from [this line of code](https://github.com/opencontainers/runc/blob/2cf8d240075dd322b9385100c9af4b149c973391/libcontainer/rootfs_linux.go#L325) where it checks for symlinks.

If I hijack the task and run the following commands:
```
/tmp/buildkitd/runc-overlayfs/snapshots/snapshots/42/fs # rm -rf proc
/tmp/buildkitd/runc-overlayfs/snapshots/snapshots/42/fs # mkdir -p proc
/tmp/buildkitd/runc-overlayfs/snapshots/snapshots/42/fs # rm -rf sys
/tmp/buildkitd/runc-overlayfs/snapshots/snapshots/42/fs # mkdir -p sys
```
I can then manually execute the `build` command which then succeeds.

**Comment by vito** [comment]
Not sure what would be causing this - would likely need to know how your workers are deployed.

Also can you try again with v0.5? It has a fix for #50 which I suspect affects more things than performance.

**Comment by sneal** [comment]
Our image definition has changed pretty substantially since I originally open this issue. I'll re-open this issue we encounter this again.
